src/js/libcs/Operators.js

```diff
@@ -4109,11 +4118,101 @@ evaluator.create$3 = function(args, modifs) {
     if (a.length > 0)
         el.args = a;
 
-    var index = evaluateAndVal(modifs.index);
-    if (index.ctype === "number")
-        el.index = index.value.real | 0;
+    for (var field in emodifs) {
+        el[field] = General.unwrap(emodifs[field]);
+    }
+
+    return {
+        'ctype': 'geo',
+        'value': addElement(el)
+    };
+};
+
+evaluator.create$2 = function(args, modifs) {
+    var type = evaluate(args[0]);
+    var defs = evaluate(args[1]);
+    var emodifs = {};
+    for (var key in modifs) {
+        emodifs[key] = evaluateAndVal(modifs[key]);
+    }
+
+    if (!geoOps.hasOwnProperty(type.value) &&
+        !geoAliases.hasOwnProperty(type.value) &&
+        !geoMacros.hasOwnProperty(type.value)) {
+        printStackTrace("Invalid geometric operation: '" + type.value + "'");
+        return nada;
+    }
+
+    // Recursively apply aliases
+    while (geoAliases.hasOwnProperty(type.value)) {
+        type.value = geoAliases[type.value];
+    }
+
+    // Detect unsupported operations or missing or incorrect arguments
+    var op = geoOps[type.value];
+
+
+    function getFirstFreeName(kind) {
+        var ans = false;
+
+        function useiffree(name) {
+            if (!csgeo.csnames[name]) {
+                ans = name;
+            }
+        }
+        var name, i;
+        if (kind === 'P') {
```

Comment:
This lookup seems to be fine. Perhaps we can later change the whole name finding process to JavaScript generators: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

Reply:
The original lookup in Cinderella skips I an J as point names (for good reason). That should happen here as well.

Reply:
Thank you. This issue is now dealt with in #671